modules/build/src/main/scala/scala/build/bsp/protocol/WorkspaceEdit.scala

```diff
@@ -0,0 +1,18 @@
+package scala.build.bsp.protocol
+
+import ch.epfl.scala.bsp4j as b
+
+/** A WorkspaceEdit meant to mimic the LSP WorkspaceEdit.
+  *
+  * https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspaceEdit
+  *
+  * @param changes
+  *   A Map of the URI that contains the diagnostic to the TextEdits that it contains.
+  * @param documentChanges
+  *   An array of resource operations that can just be an array of TextDocumentEdit[] or the various
+  *   file resource operation.
+  */
+case class WorkspaceEdit(
+  changes: Map[String, Array[TextEdits.TextEdit]],
+  documentChanges: Array[ResourceOperation] = Array.empty
```

Comment:
Why not to use `WorkspaceEdit` from `org.eclipse.lsp4j`? and I'm fine with adding new dependency `org.eclipse.lsp4j` to `scala-cli` project

cc: @tgodzik

Reply:
Ha, I originally thought of that since it's a but silly to have to mimic all that, but also noticed that the specific needed lsp4j package wasn't in scope here. If we can add it, then yea this makes all of this way easier.

Reply:
Sure, we can add it. This won't increase the significant size of the scala-cli launcher.

Reply:
Ok cool, I'll update this then to go that route... should have asked first 😬